infra/main.bicep

```diff
@@ -104,49 +113,45 @@ module machineLearningEndpoint './core/host/ml-online-endpoint.bicep' = {
   }
 }
 
-module userAcrRolePush 'core/security/role.bicep' =
-  if (!empty(principalId)) {
-    name: 'user-acr-role-push'
-    scope: rg
-    params: {
-      principalId: principalId
-      roleDefinitionId: '8311e382-0749-4cb8-b61a-304f252e45ec'
-      principalType: 'User'
-    }
+module userAcrRolePush 'core/security/role.bicep' = if (!empty(principalId)) {
+  name: 'user-acr-role-push'
+  scope: rg
+  params: {
+    principalId: principalId
+    roleDefinitionId: '8311e382-0749-4cb8-b61a-304f252e45ec'
+    principalType: principalType
   }
+}
 
-module userAcrRolePull 'core/security/role.bicep' =
-  if (!empty(principalId)) {
-    name: 'user-acr-role-pull'
-    scope: rg
-    params: {
-      principalId: principalId
-      roleDefinitionId: '7f951dda-4ed3-4680-a7ca-43fe172d538d'
-      principalType: 'User'
-    }
+module userAcrRolePull 'core/security/role.bicep' = if (!empty(principalId)) {
+  name: 'user-acr-role-pull'
+  scope: rg
+  params: {
+    principalId: principalId
+    roleDefinitionId: '7f951dda-4ed3-4680-a7ca-43fe172d538d'
+    principalType: principalType
   }
+}
 
-module userRoleDataScientist 'core/security/role.bicep' =
-  if (!empty(principalId)) {
-    name: 'user-role-data-scientist'
-    scope: rg
-    params: {
-      principalId: principalId
-      roleDefinitionId: 'f6c7c914-8db3-469d-8ca1-694a8f32e121'
-      principalType: 'User'
-    }
+module userRoleDataScientist 'core/security/role.bicep' = if (!empty(principalId)) {
```

Comment:
Just wondering... are these required for a "deployer app" to provision the infrastructure, or would we only want to assign it for local users?

Reply:
In this template these would only be valid for actual users that log into the AI workspace.